core/src/avm1/globals/color_transform.rs

```diff
@@ -0,0 +1,299 @@
+        b_add: blue_offset,
+        a_add: alpha_offset,
+    })
+}
+
+pub fn get_rgb<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error<'gc>> {
+    let ct = object_to_color_transform(this, avm, context)?;
+    let rgb = ((ct.r_add as u32) << 16) | ((ct.g_add as u32) << 8) | (ct.b_add as u32);
+    Ok(Value::Number(rgb.into()).into())
+}
+
+pub fn set_rgb<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error<'gc>> {
+    let new_rgb = args
+        .get(0)
+        .unwrap_or(&Value::Undefined)
+        .coerce_to_f64(avm, context)? as u64;
+
+    let red = ((new_rgb >> 16) & 0xFF) as f64;
+    let green = ((new_rgb >> 8) & 0xFF) as f64;
+    let blue = (new_rgb & 0xFF) as f64;
+
+    this.set("redOffset", red.into(), avm, context)?;
+    this.set("greenOffset", green.into(), avm, context)?;
+    this.set("blueOffset", blue.into(), avm, context)?;
+
+    this.set("redMultiplier", 0.into(), avm, context)?;
+    this.set("greenMultiplier", 0.into(), avm, context)?;
+    this.set("blueMultiplier", 0.into(), avm, context)?;
+
+    Ok(Value::Undefined.into())
+}
+
+pub fn create_color_transform_object<'gc>(
+    gc_context: MutationContext<'gc, '_>,
+    point_proto: Option<Object<'gc>>,
+    fn_proto: Option<Object<'gc>>,
+) -> Object<'gc> {
+    FunctionObject::function(
+        gc_context,
+        Executable::Native(constructor),
+        fn_proto,
+        point_proto,
+    )
+}
+
+pub fn create_proto<'gc>(
+    gc_context: MutationContext<'gc, '_>,
+    proto: Object<'gc>,
+    fn_proto: Object<'gc>,
+) -> Object<'gc> {
+    let mut object = ScriptObject::object(gc_context, Some(proto));
+
+    object.force_set_function(
+        "concat",
+        concat,
+        gc_context,
+        EnumSet::empty(),
+        Some(fn_proto),
+    );
+
+    object.force_set_function(
+        "toString",
+        to_string,
+        gc_context,
+        EnumSet::empty(),
+        Some(fn_proto),
+    );
+
+    object.add_property(
+        gc_context,
+        "rgb",
+        Executable::Native(get_rgb),
+        Some(Executable::Native(set_rgb)),
+        EnumSet::empty(),
+    );
+
+    object.into()
+}
+
+fn to_string<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error<'gc>> {
+    let ct = object_to_color_transform(this, avm, context)?;
+
+    let formatted = format!("(redMultiplier={}, greenMultiplier={}, blueMultiplier={}, alphaMultiplier={}, redOffset={}, greenOffset={}, blueOffset={}, alphaOffset={})",
+            ct.r_mult,
+            ct.g_mult,
+            ct.b_mult,
+            ct.a_mult,
+            ct.r_add,
+            ct.g_add,
+            ct.b_add,
+            ct.a_add
```

Comment:
This should probably call `coerce_to_string` on all of these to get the proper numeric formatting.